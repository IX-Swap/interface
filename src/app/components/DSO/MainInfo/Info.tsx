import { Grid, Typography } from '@mui/material'
import React from 'react'

export interface InfoProps {
  label: string
  value?: string | number
}
export const Info = ({ label, value }: InfoProps) => {
  return (
    <Grid container spacing={0} justifyContent='space-between'>
      <Grid itemScope>
        <Typography variant='subtitle1' color='textSecondary'>
          {label}
        </Typography>
      </Grid>
      <Grid item>
        <Typography variant='subtitle1' textAlign='right'>
          {value !== undefined ? value : '-'}
        </Typography>
      </Grid>
    </Grid>
  )
}
