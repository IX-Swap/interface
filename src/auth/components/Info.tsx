import React from 'react'
import { Grid, Typography } from '@mui/material'
import CheckIcon from '@mui/icons-material/Check'
import { green } from '@mui/material/colors'

export interface InfoProps {
  title: string
  info: string[]
}

export const Info = ({ title, info }: InfoProps) => {
  return (
    <Grid container direction='column' spacing={1}>
      <Grid item>
        <Typography variant='subtitle1' style={{ fontWeight: 500 }}>
          {title}
        </Typography>
      </Grid>
      {info.map((value, i) => (
        <Grid item key={i}>
          <Grid container spacing={1}>
            <Grid item>
              <CheckIcon style={{ color: green[500] }} />
            </Grid>
            <Grid item>
              <Typography variant='body1'>{value}</Typography>
            </Grid>
          </Grid>
        </Grid>
      ))}
    </Grid>
  )
}
