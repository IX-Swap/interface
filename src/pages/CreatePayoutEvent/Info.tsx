import React from 'react'
import { TYPE } from 'theme'
import { PayoutHeaderCard } from './styleds'

export const Info = () => {
  return (
    <PayoutHeaderCard marginTop="16px" marginBottom="32px">
      <TYPE.body style={{ fontWeight: '600' }}>What is a claim payout event?</TYPE.body>
      <TYPE.description8>Users need to manually claim payout</TYPE.description8>
    </PayoutHeaderCard>
  )
}
