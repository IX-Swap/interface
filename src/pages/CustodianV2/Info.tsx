import React from 'react'
import { Trans } from '@lingui/macro'
import styled from 'styled-components'

export const Info = () => {
  return (
    <Container>
      <div>
        <Trans>
          To trade/swap Security Tokens make sure that you have passed issuer accreditation. To do this, click on the
          desired Security Token below.
        </Trans>
      </div>
      <div>
        <Trans>
          You need to complete this process for each Security Tokens you would like to trade. The Accreditation is being
          verified by the Custodian and can take up to 1-3 days.
        </Trans>
      </div>
    </Container>
  )
}

const Container = styled.div`
  margin-top: 32px;
  margin-bottom: 40px;
  font-size: 12px;
  line-height: 18px;
  color: ${({ theme }) => theme.text2};
  display: flex;
  flex-direction: column;
  row-gap: 4px;
`
