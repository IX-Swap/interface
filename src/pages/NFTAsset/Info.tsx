import React from 'react'
import styled from 'styled-components'

interface Props {
  name: string
  collectionName: string
  description?: string
  goToCollection: () => void
}

export const Info = ({ name, collectionName, description, goToCollection }: Props) => {
  return (
    <Container>
      <div onClick={goToCollection}>{collectionName}</div>
      <div>{name}</div>
      {description && <div>{description}</div>}
    </Container>
  )
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  row-gap: 12px;
  > div:nth-child(1) {
    font-size: 14px;
    text-decoration: underline;
    cursor: pointer;
  }
  > div:nth-child(2) {
    font-weight: 600;
    font-size: 36px;
  }
  > div:nth-child(3) {
    font-size: 18px;
  }
`
