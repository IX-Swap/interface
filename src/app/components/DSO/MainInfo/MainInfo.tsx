import { Grid, Paper, Typography } from '@mui/material'
import { Info } from 'app/components/DSO/MainInfo/Info'
import { Expandable } from 'app/components/Expandable/Expandable'
import { formatDateToMMDDYY } from 'helpers/dates'
import { useAppBreakpoints } from 'hooks/useAppBreakpoints'
import React from 'react'
import { DigitalSecurityOffering } from 'types/dso'

export interface MainInfoProps {
  dso: DigitalSecurityOffering
}

export const MainInfo = ({ dso }: MainInfoProps) => {
  const { isTablet } = useAppBreakpoints()

  const expandedComponent = () => (
    <>
      <Grid item xs={12}>
        <Info label='Network' value={dso.network?.name} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Decimal' value={dso.decimals} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Capital Structure' value={dso.capitalStructure} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Launch Date' value={formatDateToMMDDYY(dso.launchDate)} />
      </Grid>
      <Grid item xs={12}>
        <Info
          label='Completion Date'
          value={formatDateToMMDDYY(dso.completionDate)}
        />
      </Grid>
    </>
  )

  return (
    <Paper sx={{ p: { xs: 3, md: 5 }, borderRadius: 2 }}>
      {isTablet ? (
        <Expandable
          noBorders
          spacing={3}
          showArrow
          mainComponent={<Typography variant='h4'>Main Info</Typography>}
          expandedComponent={
            <Grid container spacing={3}>
              {expandedComponent()}
            </Grid>
          }
        />
      ) : (
        <Grid container spacing={3}>
          <Grid item xs={12}>
            <Typography variant='h4'>Main Info</Typography>
          </Grid>
          {expandedComponent()}
        </Grid>
      )}
    </Paper>
  )
}
