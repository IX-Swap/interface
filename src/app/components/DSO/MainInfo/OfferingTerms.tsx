import { Grid, Paper, Typography } from '@mui/material'
import { Info } from 'app/components/DSO/MainInfo/Info'
import { Expandable } from 'app/components/Expandable/Expandable'
import { getPercentageValue } from 'config/utils'
import { useAppBreakpoints } from 'hooks/useAppBreakpoints'
import React from 'react'
import { DigitalSecurityOffering } from 'types/dso'

export interface OfferingTermsProps {
  dso: DigitalSecurityOffering
}

export const OfferingTerms = ({ dso }: OfferingTermsProps) => {
  const { isTablet } = useAppBreakpoints()
  const dividendYield =
    dso.dividendYield !== undefined
      ? `${getPercentageValue(dso.dividendYield)}%`
      : undefined
  const interestRate =
    dso.interestRate !== undefined
      ? `${getPercentageValue(dso.interestRate)}%`
      : undefined
  const grossIRR =
    dso.grossIRR !== undefined
      ? `${getPercentageValue(dso.grossIRR)}%`
      : undefined

  const equityMultiple =
    dso.equityMultiple !== undefined
      ? `${getPercentageValue(dso.equityMultiple)}%`
      : undefined

  const investmentPeriod =
    dso.investmentPeriod !== undefined
      ? `${dso.investmentPeriod} ${
          dso.investmentPeriod === 1 ? 'Month' : 'Months'
        }`
      : undefined

  const expandedComponent = () => (
    <>
      <Grid item xs={12}>
        <Info label='Investment Period' value={investmentPeriod} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Dividend Yield' value={dividendYield} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Interest Rate' value={interestRate} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Investment Structure' value={dso.investmentStructure} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Gross IRR' value={grossIRR} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Leverage' value={dso.leverage} />
      </Grid>
      <Grid item xs={12}>
        <Info label='Equity Multiple' value={equityMultiple} />
      </Grid>
      <Grid item xs={12}>
        <Info
          label='Distribution Frequency'
          value={dso.distributionFrequency}
        />
      </Grid>
    </>
  )
  return (
    <Paper sx={{ p: { xs: 3, md: 5 }, borderRadius: 2 }}>
      {isTablet ? (
        <Expandable
          noBorders
          spacing={3}
          showArrow
          mainComponent={<Typography variant='h4'>Offering Terms</Typography>}
          expandedComponent={
            <Grid container spacing={3}>
              {expandedComponent()}
            </Grid>
          }
        />
      ) : (
        <Grid container spacing={3}>
          <Grid item xs={12}>
            <Typography variant='h4'>Offering Terms</Typography>
          </Grid>
          {expandedComponent()}
        </Grid>
      )}
    </Paper>
  )
}
