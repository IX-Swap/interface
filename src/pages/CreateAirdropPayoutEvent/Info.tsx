import { PayoutHeaderCard } from 'pages/CreatePayoutEvent/styleds'
import React from 'react'
import { TYPE } from 'theme'

export const Info = () => {
  return (
    <PayoutHeaderCard marginTop="16px" marginBottom="32px">
      <TYPE.body style={{ fontWeight: '600' }}>What is an airdrop payout event?</TYPE.body>
      <TYPE.description8>Send tokens directly to users</TYPE.description8>
    </PayoutHeaderCard>
  )
}
