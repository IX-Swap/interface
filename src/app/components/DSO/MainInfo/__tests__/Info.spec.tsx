import { Info } from 'app/components/DSO/MainInfo/Info'
import React from 'react'
import { render, cleanup } from 'test-utils'

describe('Info', () => {
  afterEach(async () => {
    await cleanup()
    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    const { container } = render(<Info label='My Info' value={123456} />)
    expect(container).toMatchSnapshot()
  })
})
