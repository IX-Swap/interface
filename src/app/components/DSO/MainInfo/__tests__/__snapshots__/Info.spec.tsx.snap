// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
      itemscope=""
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-3phns2-MuiTypography-root"
      >
        My Info
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-1fjywxu-MuiTypography-root"
      >
        123456
      </h6>
    </div>
  </div>
</div>
`;
